canonical/juju-dashboard
# WD-6278 - Fix additional JIMM controllers

## Done

- Add an authentication flow for additional controllers that use JIMM.

## QA

- Connect your dashboard to a local controller.
- Go to the controllers page.
- Click the "Register a controller" button.
- Fill out the form using these values:
  - Host: `jimm.jujucharms.com`
  - This controller uses an external identity provider: true
- Submit the form.
- The controller should get added, but the status will be "Authentication needed".
- A notification should appear at the top of the page as well as a pop up notification at the bottom right (this is so that it will appear if you're on a different page when this appears.
- Click the button/link in one of the notifications.
- In the tab that opens go through the SSO process and close the tab once complete.
- The new controller should now say "Connected" (it's possible that it might say error if the login took too long, if it does try refreshing the window).
- Refresh the window and the controller should authenticate and connect again automatically.

## Details

- https://warthogs.atlassian.net/browse/WD-6278
- Fixes: #1630.

## Screenshots

![Screen Recording 2023-09-14 at 3 19 50 pm](https://github.com/canonical/juju-dashboard/assets/361637/b2d293ab-51cb-4363-a645-e8da0174b5e3)
